### PR TITLE
fix(workspace): fix glob pattern detection

### PR DESCRIPTION
**Pull request by hi-ogawa**
### Description

- closes https://github.com/vitest-dev/vitest/issues/6499

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: packages/vitest/src/node/workspace/resolveWorkspace.ts [hunk 1]
```diff
@@ -158,7 +158,7 @@ async function resolveWorkspaceProjectConfigs(
       const stringOption = definition.replace('<rootDir>', vitest.config.root)
       // if the string doesn't contain a glob, we can resolve it directly
       // ['./vitest.config.js']
-      if (!stringOption.includes('*')) {
+      if (!stringOption.includes('*') && !stringOption.startsWith('!')) {
```
Comment (hi-ogawa) [inline packages/vitest/src/node/workspace/resolveWorkspace.ts:161]: This should probably use `isDynamicPattern` from fast-glob https://github.com/mrmlnc/fast-glob?tab=readme-ov-file#isdynamicpatternpattern-options, but it looks like it's not supported yet on tinyglobby https://github.com/SuperchupuDev/tinyglobby/issues/28

**Comment by timbo-dev** [comment]
Any predictions for the conclusion of this problem?

File: packages/vitest/src/node/workspace/resolveWorkspace.ts (hunk 1 above)
Comment (sheremet-va) [reply packages/vitest/src/node/workspace/resolveWorkspace.ts:161]: Can we copy paste it maybe? Other cases includes {} for example

Comment (hi-ogawa) [reply packages/vitest/src/node/workspace/resolveWorkspace.ts]: Sounds good to me. I copied `isDynamicPattern` from fast-glob.

**Review by sheremet-va — APPROVED** [review]

**Comment by SuperchupuDev** [comment]
hi @hi-ogawa! i saw `isDynamicPattern` mentioned here, and i just wanted to let you know that today's `tinyglobby` release added support for it, meaning you don't need to copy the implementation from `fast-glob` anymore :-)

**Pull request merged**